### ISSUE TITLE
Doesn't work with pnpm

### ISSUE THREAD

**Issue by flybayer** [blitz-js/blitz#1372]
### What is the problem?

pnpm works fine with Next.js via next@canary. However in blitz apps, it fails with:

```
$ pnpm start

> a1@1.0.0 start /private/tmp/a1
> blitz start

You are using alpha software - if you have any problems, please open an issue here:
      https://github.com/blitz-js/blitz/issues/new/choose

✔ Prepped for launch
Loaded env from /private/tmp/a1/.blitz/caches/dev/.env
warn  - You have enabled experimental feature(s).
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use them at your own risk.

ready - started server on http://localhost:3000
error - /private/tmp/a1/node_modules/.pnpm/next@9.5.6-canary.9_e443920b660c5837ee38c1b88614b5b4/node_modules/next/dist/client/next-dev.js
Error: Cannot find module 'next/babel' from '/private/tmp/a1/.blitz/caches/dev'
```

### Steps to Reproduce

1. Create new blitz app (`blitz new myapp`)
1. Add this to package.json
```
  "resolutions": {
    "next": "9.5.6-canary.9"
  },
```
1. `rm -r node_modules`
1. `pnpm install`

**Comment by flybayer** [comment]
A second issue is using the global blitz cli. This is with blitz installed globally with pnpm and with app deps installed with pnpm.

```
blitz s
You are using alpha software - if you have any problems, please open an issue here:
      https://github.com/blitz-js/blitz/issues/new/choose

internal/modules/cjs/loader.js:968
  throw err;
  ^

[Error: Cannot find module 'react-dom'
Require stack:
- /Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/react-query@2.23.0/node_modules/react-query/lib/react/reactBatchedUpdates.js
- /Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/react-query@2.23.0/node_modules/react-query/lib/index.js
- /Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/@blitzjs/core@0.24.3/node_modules/@blitzjs/core/dist/core.cjs.development.js
- /Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/@blitzjs/core@0.24.3/node_modules/@blitzjs/core/dist/index.js
- /Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/@blitzjs/server@0.24.3/node_modules/@blitzjs/server/dist/server.cjs.development.js
- /Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/@blitzjs/server@0.24.3/node_modules/@blitzjs/server/dist/index.js
- /private/tmp/a1/blitz.config.js
- /private/tmp/a1/node_modules/.pnpm/@blitzjs/config@0.24.3/node_modules/@blitzjs/config/dist/config.cjs.development.js
- /private/tmp/a1/node_modules/.pnpm/@blitzjs/config@0.24.3/node_modules/@blitzjs/config/dist/index.js
- /private/tmp/a1/node_modules/.pnpm/@blitzjs/core@0.24.3_8f92a3d82ba799f00ce0d4ee95ee72c3/node_modules/@blitzjs/core/dist/core.cjs.development.js
- /private/tmp/a1/node_modules/.pnpm/@blitzjs/core@0.24.3_8f92a3d82ba799f00ce0d4ee95ee72c3/node_modules/@blitzjs/core/dist/index.js
- /private/tmp/a1/node_modules/.pnpm/@blitzjs/server@0.24.3_e443920b660c5837ee38c1b88614b5b4/node_modules/@blitzjs/server/dist/server.cjs.development.js
- /private/tmp/a1/node_modules/.pnpm/@blitzjs/server@0.24.3_e443920b660c5837ee38c1b88614b5b4/node_modules/@blitzjs/server/dist/index.js
- /private/tmp/a1/node_modules/.pnpm/@blitzjs/cli@0.24.3_typescript@4.0.3/node_modules/@blitzjs/cli/lib/src/commands/start.js
- /private/tmp/a1/node_modules/.pnpm/@oclif/config@1.15.1/node_modules/@oclif/config/lib/plugin.js
- /private/tmp/a1/node_modules/.pnpm/@oclif/config@1.15.1/node_modules/@oclif/config/lib/config.js
- /private/tmp/a1/node_modules/.pnpm/@oclif/config@1.15.1/node_modules/@oclif/config/lib/index.js
- /private/tmp/a1/node_modules/.pnpm/@oclif/command@1.5.20/node_modules/@oclif/command/lib/command.js
- /private/tmp/a1/node_modules/.pnpm/@oclif/command@1.5.20/node_modules/@oclif/command/lib/index.js
- /private/tmp/a1/node_modules/.pnpm/@blitzjs/cli@0.24.3_typescript@4.0.3/node_modules/@blitzjs/cli/lib/src/index.js
- /Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/blitz@0.24.3/node_modules/blitz/dist/cli.js] {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/react-query@2.23.0/node_modules/react-query/lib/react/reactBatchedUpdates.js',
    '/Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/react-query@2.23.0/node_modules/react-query/lib/index.js',
    '/Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/@blitzjs/core@0.24.3/node_modules/@blitzjs/core/dist/core.cjs.development.js',
    '/Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/@blitzjs/core@0.24.3/node_modules/@blitzjs/core/dist/index.js',
    '/Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/@blitzjs/server@0.24.3/node_modules/@blitzjs/server/dist/server.cjs.development.js',
    '/Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/@blitzjs/server@0.24.3/node_modules/@blitzjs/server/dist/index.js',
    '/private/tmp/a1/blitz.config.js',
    '/private/tmp/a1/node_modules/.pnpm/@blitzjs/config@0.24.3/node_modules/@blitzjs/config/dist/config.cjs.development.js',
    '/private/tmp/a1/node_modules/.pnpm/@blitzjs/config@0.24.3/node_modules/@blitzjs/config/dist/index.js',
    '/private/tmp/a1/node_modules/.pnpm/@blitzjs/core@0.24.3_8f92a3d82ba799f00ce0d4ee95ee72c3/node_modules/@blitzjs/core/dist/core.cjs.development.js',
    '/private/tmp/a1/node_modules/.pnpm/@blitzjs/core@0.24.3_8f92a3d82ba799f00ce0d4ee95ee72c3/node_modules/@blitzjs/core/dist/index.js',
    '/private/tmp/a1/node_modules/.pnpm/@blitzjs/server@0.24.3_e443920b660c5837ee38c1b88614b5b4/node_modules/@blitzjs/server/dist/server.cjs.development.js',
    '/private/tmp/a1/node_modules/.pnpm/@blitzjs/server@0.24.3_e443920b660c5837ee38c1b88614b5b4/node_modules/@blitzjs/server/dist/index.js',
    '/private/tmp/a1/node_modules/.pnpm/@blitzjs/cli@0.24.3_typescript@4.0.3/node_modules/@blitzjs/cli/lib/src/commands/start.js',
    '/private/tmp/a1/node_modules/.pnpm/@oclif/config@1.15.1/node_modules/@oclif/config/lib/plugin.js',
    '/private/tmp/a1/node_modules/.pnpm/@oclif/config@1.15.1/node_modules/@oclif/config/lib/config.js',
    '/private/tmp/a1/node_modules/.pnpm/@oclif/config@1.15.1/node_modules/@oclif/config/lib/index.js',
    '/private/tmp/a1/node_modules/.pnpm/@oclif/command@1.5.20/node_modules/@oclif/command/lib/command.js',
    '/private/tmp/a1/node_modules/.pnpm/@oclif/command@1.5.20/node_modules/@oclif/command/lib/index.js',
    '/private/tmp/a1/node_modules/.pnpm/@blitzjs/cli@0.24.3_typescript@4.0.3/node_modules/@blitzjs/cli/lib/src/index.js',
    '/Users/b/.fnm/node-versions/v12.18.3/installation/pnpm-global/4/node_modules/.pnpm/blitz@0.24.3/node_modules/blitz/dist/cli.js'
  ]
}
```

**Comment by zkochan** [comment]
Regarding the initial issue, looks like the bundler or whatnot is searching for the next dependency in the root of node_modules, so either install next as a direct dep, or fix the bundler or use `public-hoist-pattern[]=next`

**Comment by abuuzayr** [comment]
I can recreate this issue. I followed the steps to reproduce, ran `pnpm start` and got the same error message. 

I also tried @zkochan's suggestion and added `public-hoist-pattern[]=next` to `.npmrc`, but I got the same error. 

I even tried adding `public-hoist-pattern[]=next/babel` instead, to no avail. 

Perhaps I did it wrong? 

At the same time, I did notice that there is no `.next` folder in `.blitz/caches/dev`, unlike if I ran `blitz start` in a regular blitz project not installed by yarn/npm. Could this be an issue with the bundler as similarly suggested?
